src/backend/distributed/executor/subplan_execution.c

```diff
@@ -43,13 +48,22 @@ ExecuteSubPlans(DistributedPlan *distributedPlan)
 
 		char *resultId = GenerateResultId(planId, subPlanId);
 
+		if (MaxIntermediateResult >= 0)
+		{
+			SubPlanLevel++;
```

Comment:
I think this variable might have other uses and increasing/decreasing it costs almost nothing. Maybe remove the `if`